streaming/index.js

```diff
@@ -17,8 +18,13 @@ const WebSocket = require('ws');
 
 const environment = process.env.NODE_ENV || 'development';
 
+// Correctly detect .env/.env.production files based on whether we're running
+// from the streaming/ directory or the root of the mastodon project.
+const dotenvFile = environment === 'production' ? '.env.production' : '.env';
+const dotenvDir = path.basename(process.cwd()) === 'streaming' ? `../` : './';
+
 dotenv.config({
-  path: environment === 'production' ? '.env.production' : '.env',
+  path: path.resolve(dotenvDir, dotenvFile)
```

Comment:
Could we rather load it with a relative path from `index.js`? This seems cleaner

```
path: `${__dirname}/../${dotenvFile}`
```

Reply:
You could, but that'd pose problems if we were to migrate to ESM, as `__dirname` isn't easily available in ESM, additionally, you don't want to use `/../` as that could break non-linux systems (that use a different path separator).

Reply:
If we manage to switch to ESM, we can still handle this properly using `new URL(`../${dotenvFile}`, import.meta.url)`

For my example above, you can use `path.join` or `path.resolve` to work around it

Reply:
Also this implementation is less error-prone, for example if the user launches the streaming process without the correct cwd (like it could happen when launched by systemd).
Referencing the `.env` file using the source-code's path is consistent with the way it is done in Ruby (`dotenv-ruby` uses `Rails.root` for this)